generators/app/templates/Gulpfile.js

```diff
@@ -37,19 +52,16 @@ function test () {
         }));
 }
 
-function preview () {
-    var buildReporterPlugin = require('testcafe').embeddingUtils.buildReporterPlugin;
-    var pluginFactory       = require('./lib');
-    var reporterTestCalls   = require('./test/utils/reporter-test-calls');
-    var plugin              = buildReporterPlugin(pluginFactory);
+async function preview () {
+    const createReport = require('./test/utils/create-report');
 
-    console.log();
+    const reportWithColors = await createReport(true);
+    const reportWithoutColors = await createReport(false);
 
-    reporterTestCalls.forEach(function (call) {
-        plugin[call.method].apply(plugin, call.args);
-    });
+    fs.writeFileSync('report-with-colors.json', JSON.stringify(reportWithColors));
+    fs.writeFileSync('report-without-colors', reportWithoutColors);
 
-    process.exit(0);
+    console.log(reportWithColors);
```

Comment:
I' thought if we could simplify the preview task. I think it can be controlled via CLI keys:
 - `--colors`/`--no-colors` for coloring (since TestCafe uses [chalk](https://github.com/chalk/chalk) for coloring,  it's a kind-of built-in key - see [here](https://github.com/chalk/chalk/tree/48905d08052aad4c8ba53bbd9fbcd8a9faf4f6e5#chalksupportscolor))
 - `--file` or `--out` for redirecting a preview to a file.

If you are interested, let's discuss it in Teams.

Reply:
After the conversation we decided that for the sake of simplicity:
- `gulp preview` (preview-no-colors + preview-colors) prints to the stdout;
- `gulp generateTestData` (generateTestDataNoColors + generateTestDataColors) prints to the corresponding files in `test/data` dir.